pkg/liqo-controller-manager/podstatus-controller/podstatus_controller_test.go

```diff
@@ -0,0 +1,215 @@
+		r := &PodStatusReconciler{
+			Client:          fakeClient,
+			Scheme:          scheme.Scheme,
+			LocalPodsClient: fakeLocalPodsClient,
+		}
+		_, err = r.Reconcile(ctx, reqNode)
+		Expect(err).NotTo(HaveOccurred())
+		klog.Flush()
+	})
+
+	When("liqo node not ready", func() {
+		BeforeEach(func() {
+			fakeClient = fakeClientBuilder.WithObjects(newLiqoNode(false), newPod()).Build()
+		})
+
+		When("local status label not present", func() {
+			BeforeEach(func() {
+				fakeLocalPodsClient = fakeClientBuilder.WithObjects(newLocalPod()).Build()
+			})
+
+			It("should add local status label to local offloaded pods", func() {
+				localPod := corev1.Pod{}
+				Expect(fakeLocalPodsClient.Get(ctx, reqLocalPod.NamespacedName, &localPod)).To(Succeed())
+				Expect(localPod.Labels).To(HaveKeyWithValue(consts.PodStatusLocalKey, consts.PodStatusLocalValue))
+			})
+		})
+
+		When("local status label present", func() {
+			BeforeEach(func() {
+				localPod := newLocalPod()
+				localPod.Labels = labels.Merge(localPod.Labels, labels.Set{consts.PodStatusLocalKey: consts.PodStatusLocalValue})
+				fakeLocalPodsClient = fakeClientBuilder.WithObjects(localPod).Build()
+			})
+
+			It("should keep local status label to local offloaded pods", func() {
+				localPod := corev1.Pod{}
+				Expect(fakeLocalPodsClient.Get(ctx, reqLocalPod.NamespacedName, &localPod)).To(Succeed())
+				Expect(localPod.Labels).To(HaveKeyWithValue(consts.PodStatusLocalKey, consts.PodStatusLocalValue))
+			})
+		})
+	})
+
+	When("liqo node ready", func() {
+		BeforeEach(func() {
+			fakeClient = fakeClientBuilder.WithObjects(newLiqoNode(true), newPod()).Build()
+		})
+
+		When("local status label not present", func() {
+			BeforeEach(func() {
+				fakeLocalPodsClient = fakeClientBuilder.WithObjects(newLocalPod()).Build()
+			})
+
+			It("should not add local status label ro local offloaded pods", func() {
+				localPod := corev1.Pod{}
+				Expect(fakeLocalPodsClient.Get(ctx, reqLocalPod.NamespacedName, &localPod)).To(Succeed())
+				Expect(localPod.Labels).ToNot(HaveKeyWithValue(consts.PodStatusLocalKey, consts.PodStatusLocalValue))
+			})
+		})
+
+		When("local status label present", func() {
+			BeforeEach(func() {
+				localPod := newLocalPod()
+				localPod.Labels = labels.Merge(localPod.Labels, labels.Set{consts.PodStatusLocalKey: consts.PodStatusLocalValue})
+				fakeLocalPodsClient = fakeClientBuilder.WithObjects(localPod).Build()
+			})
+
+			It("should remove local status label to local offloaded pods", func() {
+				localPod := corev1.Pod{}
+				Expect(fakeLocalPodsClient.Get(ctx, reqLocalPod.NamespacedName, &localPod)).To(Succeed())
+				Expect(localPod.Labels).ToNot(HaveKeyWithValue(consts.PodStatusLocalKey, consts.PodStatusLocalValue))
+			})
+		})
+
+	})
+
+	AfterEach(func() {
+		pod := corev1.Pod{}
+		Expect(fakeClient.Get(ctx, reqPod.NamespacedName, &pod)).To(Succeed())
+		Expect(pod.Labels).ToNot(HaveKeyWithValue(consts.PodStatusLocalKey, consts.PodStatusLocalValue))
```

Comment:
You are never using this client to manage pods, it is probably better to add a second pod (not selected bt the controller) and do this check on it